torch/distributed/fsdp/_wrap_utils.py

```diff
@@ -62,6 +63,13 @@ def _auto_wrap(
                 root_module, mixed_precision._module_classes_to_ignore
             )
             _warn_on_overridden_mixed_precision(overridden_module_classes)
+        use_orig_params = fsdp_kwargs.get("use_orig_params", False)
+        _validate_frozen_params(
+            root_module,
+            set(target_module_to_kwargs.keys()),
+            ignored_params,
+            use_orig_params,
+        )
         wrap_fn = _construct_wrap_fn(root_module, target_module_to_kwargs, fsdp_fn)
         _post_order_apply(root_module, wrap_fn)
```

Comment:
Note to self: it's post-order traversal because you want to wrap the children modules first, and THEN wrap the entire thing when you're all done. (This is different from the DFS validation order, so it makes sense that you need a separate traversal and can't do the error checks as you are wrapping.)

Reply:
Edit: nyeh, now I'm confused. Isn't your reverse topological sort precisely a post order traversal?